aten/src/ATen/native/TensorConversions.cpp

```diff
@@ -363,26 +363,57 @@ Tensor sparse_compressed_to_dense(
     return dst.add_(self);
   }
   if (self.layout() == kSparseBsr) {
-    TORCH_CHECK(self.dim() == 2, "Can only convert 2D SparseBsr to Strided.");
+    TORCH_CHECK(self.dim() == 2 || self.dim() == 3, "Can only convert 2D or 3D SparseBsr to Strided.");
+    auto crow_indices = self.crow_indices();
+    auto col_indices = self.col_indices();
+    if (self.dim() == 3) {
```

Comment:
We will want to add a runtime check here 
that verifies all the indices are the same. For now it's not possible to construct a batched BSR Tensor via `to_sparse_bsr` that violates this, but it's a risky assumption. I'll push a verification. Immediate follow up work includes generalizing this code to not depend on this assumption.